### PR TITLE
Add private mirror support for OCI references #4362

### PR DESCRIPTION
**Pull request by elysweyr**
Implements #4362

### CODE REVIEW

File: nextcloud-aio-helm-chart/templates/nextcloud-aio-apache-deployment.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ spec:
             - name: COLLABORA_HOST
               value: nextcloud-aio-collabora
             - name: NC_DOMAIN
-              value: "{{ .Values.NC_DOMAIN }}"
+              value: "{{ .Values.NC_DOMAIN }}"FF
```
Comment (Zoey2936) [inline nextcloud-aio-helm-chart/templates/nextcloud-aio-apache-deployment.yaml:53]: is there a reason for this "FF"?

**Review by Zoey2936 — CHANGES_REQUESTED** [review]
I think you need to edit this file: https://github.com/nextcloud/all-in-one/blob/main/nextcloud-aio-helm-chart/update-helm.sh because the helm-chart is generated using this script your changes will be removed again with the next run of this script

**Comment by elysweyr** [comment]
> I think you need to edit this file: https://github.com/nextcloud/all-in-one/blob/main/nextcloud-aio-helm-chart/update-helm.sh because the helm-chart is generated using this script your changes will be removed again with the next run of this script

@Zoey2936 
Updated `update-helm.sh` and refined my control flow in case `ALPINE_IMAGE_ORG` and `IMAGE_MIRROR_PREFIX` are unset which would have lead to `/alpine` but if fixed now.

File: nextcloud-aio-helm-chart/templates/nextcloud-aio-apache-deployment.yaml (hunk 1 above)
Comment (elysweyr) [reply nextcloud-aio-helm-chart/templates/nextcloud-aio-apache-deployment.yaml]: My bad

**Comment by szaimen** [comment]
Hi, have you tested if this actually works? Because I doubt it. See https://github.com/nextcloud/all-in-one/discussions/4362#discussioncomment-8773102

**Comment by szaimen** [comment]
Hi @elysweyr I just gave you write access to the repo and pushed your changes to https://github.com/nextcloud/all-in-one/pull/4387 so that all workflows work. Lets continue there.

**Pull request closed without merging**